lib/policy_machine/policy_element.rb

```diff
@@ -65,7 +65,9 @@ def unassign(dst_policy_element)
     # Removes self and any assignments to or from self. Does not remove any other elements.
     # Returns true if persisted object was successfully removed.
     def delete
-      if self.stored_pe && self.stored_pe.persisted
+      # If the stored ppe is not persisted, no need to delete it, but if buffering
+      # all bets are off, and the storage adapter needs to handle the situation
+      if (self.stored_pe && self.stored_pe.persisted) || pm_storage_adapter.buffering?
```

Comment:
Should we check if it's buffering first so that we don't have to make that extra db call unless we have to?